torque-core/src/main/kotlin/com/workday/torque/TestChunkRunner.kt

```diff
@@ -17,32 +19,49 @@ class TestChunkRunner(
         return try {
             adbDevice.log("Starting tests...")
             installer.ensureTestPackageInstalled(args, testChunk)
+            makeTestFileDirectories(args).await()
             runAndParseTests(args.chunkTimeoutSeconds, testChunk).await()
         } catch (e: Exception) {
             adbDevice.log("TestChunk run crashed with exception: ${e.message}")
             createCrashedAdbDeviceTestResults(testChunk, e)
         }
     }
 
+    private fun makeTestFileDirectories(args: Args): Completable {
+        return processRunner.runAdb(commandAndArgs = listOf(
+                        "-s", adbDevice.id,
+                        "shell", "mkdir -p ${args.testFilesPullDeviceDirectory}/coverage-reports"
+                ),
+                        destroyOnUnsubscribe = true)
+                .ofType(Notification.Exit::class.java)
+                .map { true }
+                .doOnError { error -> adbDevice.log("Failed to mkdir on ${adbDevice.tag}, filepath: ${args.testFilesPullDeviceDirectory}/coverage-reports, failed: $error") }
+                .ignoreElements()
+    }
+
     private fun runAndParseTests(chunkTimeoutSeconds: Long, testChunk: TestChunk): Single<List<AdbDeviceTestResult>> {
         val testPackageName = testChunk.testModuleInfo.moduleInfo.apkPackage.value
         val testRunnerClass = testChunk.testModuleInfo.testRunner.value
+        val coverageFileName = testChunk.testMethods.joinToString(",") { it.testName } + ".ec"
```

Comment:
How big can testMethods be?
It sounds risky to reach the maximum filename size (255 bytes) on linux.

Reply:
I can limit the testMethod portion of the name. Some of our test names are pretty huge, but i worry that truncation can lead to name collisions.